gapic-generator-java/src/main/java/com/google/api/generator/engine/ast/PackageInfoDefinition.java

```diff
@@ -24,6 +24,8 @@
 public abstract class PackageInfoDefinition implements AstNode {
   public abstract String pakkage();
 
+  public abstract boolean isEmpty();
+
```

Comment:
I wonder if it's possible/ makes sense, but rather than having to set a new field to represent the state (empty or not), I wonder if we can just expose a method that checks the state.

i.e. something like
```
public boolean shouldGenerateFile() {
    return !fileHeader().isEmpty() || !headerCommentStatements.isEmpty() || !annotations().isEmpty();
}
```

Reply:
Done, I modified other `isEmpty()` methods as well